crates/extensions/c8y_mapper_ext/src/operations/config_snapshot.rs

```diff
@@ -125,37 +123,93 @@ impl CumulocityConverter {
                 let config_filename =
                     format!("{}-{}", response.config_type.replace('/', ":"), cmd_id);
 
-                let tedge_file_url = format!(
+                let tedge_file_url_fallback = format!(
                     "http://{}/tedge/file-transfer/{external_id}/config_snapshot/{config_filename}",
+                    // XXX: FTS url is in tedgeUrl parameter, which we ignore, should we?
                     &self.config.tedge_http_host,
                     external_id = target.external_id.as_ref()
                 );
 
+                let tedge_file_url = response
+                    .tedge_url
+                    .as_ref()
+                    .unwrap_or(&tedge_file_url_fallback);
+
                 let destination_dir = tempfile::tempdir_in(self.config.tmp_dir.as_std_path())
                     .context("Failed to create a temporary directory")?;
                 let destination_path = destination_dir.path().join(config_filename);
 
-                self.pending_fts_download_operations.insert(
-                    cmd_id.into(),
-                    FtsDownloadOperationData {
-                        download_type: FtsDownloadOperationType::ConfigDownload,
-                        url: tedge_file_url.clone(),
-                        file_dir: destination_dir,
+                let downloader =
+                    Downloader::new(destination_path.clone(), self.config.identity.clone());
```

Comment:
I believe we can continue using the downloader actor here by using a `ClientMessageBox` to it so that we can await on the response receiver inline. The `c8y_firmware_manager` is already using the same.

Reply:
That's right, we could use the `ClientMessageBox` here, but I also wanted to explore if we could just use the downloader directly, without relying on an outside actor to do it. I'd say it has an advantage of decreasing coupling. The disadvantage is that as the messages are no longer visible, but does other code need them to be visible, other than testing code? If so, then maybe we don't need to go via the actor.

But the main way I did it like this, was to just demonstrate that after the refactor it will be possible, whereas currently this would block. But now we can easily swap this with an `EnvelopeSender` if we want to go via the actor.

Reply:
And additional benefit of going via the downloader actor is described in this comment: https://github.com/thin-edge/thin-edge.io/pull/2881#discussion_r1603970946